File: Recursion/python/fibonacci_recursive.py
class FibonacciRecursive:
    def __init__(self):
        pass

    def calculate_fibonacci(self, n: int) -> int:
        print("This program is calculating Fibonacci({}) by recursive method".format(n))
        if n <= 1:
            print("Fibonacci({}) is {}".format(n, n))
            return n
        f = self.calculate_fibonacci(n - 1) + self.calculate_fibonacci(n - 2)
        print("Fibonacci({}) is {}".format(n, f))
        return f

File: Recursion/cpp/FibonacciRecursive.cpp
#include "FibonacciRecursive.h"

FibonacciRecursive::FibonacciRecursive() {

}

int FibonacciRecursive::calculateFibonacci(const int& n) {
	cout << "This program is calculating Fibonacci(" << n << ") by recursive method.\n";
	if (n <= 1)
	{
		cout << "Fibonacci(" << n << ") is " << n << "\n";
		return n;
	}
	int f = calculateFibonacci(n - 1) + calculateFibonacci(n - 2);
	return f;
}